alacritty/src/macos/proc.rs

```diff
@@ -0,0 +1,104 @@
+use std::ffi::OsStr;
+use std::io;
+use std::mem;
+use std::os::raw;
+use std::os::unix::ffi::OsStrExt;
+use std::path::PathBuf;
+use std::slice;
+use memchr::memchr;
+
+pub fn cwd(pid: raw::c_int) -> io::Result<PathBuf> {
+    let flavor = PROC_PIDVNODEPATHINFO as raw::c_int;
+    let mut info = mem::MaybeUninit::<proc_vnodepathinfo>::uninit();
+    let size = mem::size_of::<proc_vnodepathinfo>() as raw::c_int;
+    let result = unsafe {
+        proc_pidinfo(pid, flavor, 0, info.as_mut_ptr() as *mut raw::c_void, size)
+    };
+    match result {
+        c if c <= 0    => return Err(io::Error::last_os_error()),
```

Comment:
Are you sure that `0` is an error value?

Reply:
Good question. I am quite confident it is *some* kind of error, since we simply cannot build a `proc_vnodepathinfo` without some bytes returned. That being said, I don't know for sure if it's reliable to use `last_os_error()` here, or if the other error makes more sense.

I tried looking at some code I found for the implementation of `proc_pidinfo` online, but it's not consistent with the interface we're using (it's also horrible code). Perhaps I can find the latest version somewhere...

Reply:
I'm going to move the 0 case to the `ErrorKind::Other` since I'm not at all convinced `last_os_error` will return anything meaningful, and "invalid return size" is absolutly a correct assessment of this error.